reset_password.go

```diff
@@ -104,6 +104,21 @@ func identifyAndRecover(ctx context.Context, ih *InteractionHandle, ir *Identify
 	if err != nil {
 		return nil, err
 	}
+	if resp.CurrentAuthenticatorEnrollment == nil {
```

Comment:
seems like the check on the current authenticator enrollment in `identifyAndRecover` and `VerifyEmail`  can be DRY'd up 

Have a check function like:

```go
func verifyAuthenticatorEnrollment(ctx context.Context, resp *Response) (*Response, error) {
	if resp.CurrentAuthenticatorEnrollment == nil {
		ro, authID, err := resp.authenticatorOption("select-authenticator-authenticate", "Password", true)
		if err != nil {
			return nil, err
		}
		authenticator := []byte(`{
				"authenticator": {
					"id": "` + authID + `"
				}
			}`)
		resp, err = ro.proceed(ctx, authenticator)
		if err != nil {
			return nil, err
		}
	}
	if resp.CurrentAuthenticatorEnrollment == nil {
		return nil, fmt.Errorf("falied to init password recovery: 'currentAuthenticatorEnrollment' " +
			"field is missing from the response")
	}
	return resp.CurrentAuthenticatorEnrollment.Value.Recover.proceed(ctx, nil)
}
```

and `identifyAndRecover` cutting lines 107 to 126 to have a return of 

```go
return verifyAuthenticatorEnrollment(ctx, resp) 
```

and `VerifyEmail` cut out the duplicated code on lines 137 to 156

```
resp, err = verifyAuthenticatorEnrollment(ctx, resp)
```